airbyte-integrations/connectors/source-talkdesk-explore/source_talkdesk_explore/streams.py

```diff
@@ -0,0 +1,241 @@
+        id_obj = {"id": response_json['job']['id']}
+        logger.info(f"Generated report with ID '{id_obj['id']}'")
+
+        return [id_obj]
+
+
+class ReadReportStream(HttpStream):
+    primary_key = None
+
+    def __init__(self, start_date, timezone, **kwargs):
+        super().__init__(**kwargs)
+        self.start_date = start_date
+        self.timezone = timezone
+
+    @property
+    def url_base(self) -> str:
+        return f"https://api.talkdeskapp.com/data/"
+    
+    def path(self, **kwargs) -> str:
+        latest_state = kwargs.get('stream_state').get(self.cursor_field, None)
+
+        if not latest_state:
+            latest_state = self.start_date
+        
+        # Check and set latest_state to necessary date-time format
+        try:
+            datetime.strptime(latest_state, "%Y-%m-%dT%H:%M:%S")
+        except ValueError:
+            try:
+                datetime.strptime(latest_state, "%Y-%m-%d %H:%M:%S")
+                latest_state = latest_state.replace(" ", "T")
+            except ValueError:
+                logger.error("stream_state is in unhandled date-time format. Required format: %Y-%m-%dT%H:%M:%S")
+
+        generate_report = GenerateReportStream(
+            base_path=self.base_path,
+            start_date=latest_state,
+            timezone=self.timezone,
+            authenticator=self.authenticator
+        )
+        report_id = next(generate_report.read_records(SyncMode.full_refresh))
+
+        return self.base_path + f"/{report_id['id']}"
+    
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+    
+    def should_retry(self, response: requests.Response) -> bool:
+        """
+        Retry conditions:
+        1. By default, back off on the following HTTP response statuses:
+         - 429 (Too Many Requests) indicating rate limiting
+         - 500s to handle transient server errors
+         - Unexpected but transient exceptions (connection timeout, DNS resolution failed, etc..) are retried by default.
+        2. When the report is requested but is not ready to be fetched:
+         - In that case, the response will have the following format:
+         ```
+         {"job": {"id": "369f88a5-d5a3-42c6-a135-8aec4215553e", "name": "Calls", 
+         "created_at": "2022-01-13T10:17:15", "status": "processing", "type": "calls", "format": "json", ...}}
+         ```
+         The retry function will be looking for a response in this format with 'status=processing'.
+
+        """
+        if response.status_code == 429 or 500 <= response.status_code < 600:
+            return True
+        else:
+            response_obj = response.json()
+            try:
+                report_status = response_obj["job"]["status"]
+                if report_status != "completed":
+                    logger.info("Requested report is in uncompleted status. Waiting for it to be completed...")
+                    return True
```

Comment:
Any chance to have other status like `failed`, `error`, `canceled`?

Reply:
The docs are not clear 😞 But from what I understand: when a report is successfully generated, its status will be either `processing` or `completed`. And we are only able to fetch the results when it is `completed`.

When it fails to be executed, the HTTP code of the response will be either 400, 401, 403, 429 or 503. I will implement the handling for failed reports :)